In [1]:
import pandas as pd
df = pd.read_csv("allbacks.csv")

In [2]:
# Count the Categories

print(df["cover"].value_counts())

cover
pb    8
hb    7
Name: count, dtype: int64


In [3]:
# get the middle position

(15 + 1) / 2

8.0

In [4]:
# Since pandas uses zero-based indexing, the 8th observation is at index 7

print(df["cover"].iloc[7])

pb


From what I can tell there is no good way to find the median on categories that are not numerical. I don't remember seeing it in the book nor do I remember seeing it in the lecture videos. When asking Google/AI, I also get a lot of confusing feedback. Therefore I went with a simple "get the count, find the middle number, get the value of the index at said middle number".

If there is an actual built in function for this, I would love to learn it. 

In [ ]:
# Weights of Books

print(df["weight"].mean(), "This is the average weight of all books recorded regardless of their cover type.")
mean_weight_hb = df[df["cover"] == "hb"]["weight"].mean()
print(mean_weight_hb, "This is the average weight of all the books with a hardback cover type.")
mean_weight_pb = df[df["cover"] == "pb"]["weight"].mean()
print(mean_weight_pb, "This is the average weight of all the books with a paperback cover type.")

706.6666666666666 This is the average weight of all books recorded regardless of their cover type.
796.4285714285714 This is the average weight of all the books with a hardback cover type.
628.125 This is the average weight of all the books with a paperback cover type.


The difference between the mean weights is that the hb books weigh more on average than the pb books. 

By hand verification of hb books:
800 + 950 + 1050 + 350 + 750 + 600 + 1075 = 5575
5575 / 7 = 796.43
796.43 = 796.43 check

https://github.com/skeown13/cs82a-data-science-portfolio